src/app/services/wallet.service.ts

```diff
@@ -24,6 +25,7 @@ export class WalletService implements OnDestroy {
   private rafReference: number;
   private metamaskStatusSubject = new BehaviorSubject<MetamaskStatus>(undefined);
   private walletSubject = new BehaviorSubject<Wallet>(undefined);
+  private balanceSubject = new BehaviorSubject<BigNumber>(undefined);
```

Comment:
Do we really need separate subject for `wallet` property? Wouldn't be proper to just use existing `walletSubject`?

Reply:
A lot of components subscribes for wallet address changes. These components make some requests to metaindexer and to blockchain. If I emit event on walletSubject when balance is updated all these components will unnecessary resend all requests.